src/extractor/edge_based_graph_factory.cpp

```diff
@@ -975,18 +975,24 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
                           }
                       });
 
-    // re-hash conditionals to ocnnect to their respective edge-based edges. Due to the
-    // ordering, we
-    // do not really have a choice but to index the conditional penalties and walk over all
-    // edge-based-edges to find the ID of the edge
+    // Re-hash conditionals to connect to their respective edge-based edges,
+    // by indexing the penalties and finding the IDs of each edge in the edge-based-edges
     auto const indexed_conditionals = IndexConditionals(std::move(conditionals));
+
     {
-        util::Log() << "Writing " << indexed_conditionals.size()
-                    << " conditional turn penalties...";
-        // write conditional turn penalties into the restrictions file
-        storage::io::FileWriter writer(conditional_penalties_filename,
-                                       storage::io::FileWriter::GenerateFingerprint);
-        extractor::serialization::write(writer, indexed_conditionals);
+        if (!indexed_conditionals.empty())
```

Comment:
The reason why we write out en empty file here is that otherwise the loading code will fail: https://travis-ci.org/Project-OSRM/osrm-backend/jobs/289928057#L2508

The `.restrictions` file can be empty but is not optional.